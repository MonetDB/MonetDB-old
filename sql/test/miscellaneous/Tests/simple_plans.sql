start transaction;

create table myx (x uuid, y uuid);
plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965') or y = '1aea00e5db6e0810b554fde31d961965';
plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965') or y is null;

plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965', '1aea00e5db6e0810b554fde31d961966') or y = '1aea00e5db6e0810b554fde31d961967';

insert into myx values ('1aea00e5db6e0810b554fde31d961965', '1aea00e5db6e0810b554fde31d961967');
select * from myx where x in ('1aea00e5db6e0810b554fde31d961966') or y is not null;
select * from myx where x in ('1aea00e5db6e0810b554fde31d961966') or y is null;

rollback;
