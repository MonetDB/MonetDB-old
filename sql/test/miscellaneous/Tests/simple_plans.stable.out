stdout of test 'simple_plans` in directory 'sql/test/miscellaneous` itself:


# 11:57:38 >  
# 11:57:38 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-205942" "--port=30914"
# 11:57:38 >  

#start transaction;
#create table myx (x uuid, y uuid);
#plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965') or y = '1aea00e5db6e0810b554fde31d961965';
% .plan # table_name
% rel # name
% clob # type
% 154 # length
project (
| select (
| | table(sys.myx) [ "myx"."x", "myx"."y" ] COUNT 
| ) [ ("myx"."x" in (uuid[char(32) "1aea00e5db6e0810b554fde31d961965"] as "%2"."%2")) or ("myx"."y" = uuid[char(32) "1aea00e5db6e0810b554fde31d961965"]) ]
) [ "myx"."x", "myx"."y" ]
#plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965') or y is null;
% .plan # table_name
% rel # name
% clob # type
% 118 # length
project (
| select (
| | table(sys.myx) [ "myx"."x", "myx"."y" ] COUNT 
| ) [ ("myx"."x" in (uuid[char(32) "1aea00e5db6e0810b554fde31d961965"] as "%2"."%2")) or ("myx"."y" * = uuid "NULL") ]
) [ "myx"."x", "myx"."y" ]
#plan select * from myx where x in ('1aea00e5db6e0810b554fde31d961965', '1aea00e5db6e0810b554fde31d961966') or y = '1aea00e5db6e0810b554fde31d961967';
% .plan # table_name
% rel # name
% clob # type
% 218 # length
project (
| select (
| | table(sys.myx) [ "myx"."x", "myx"."y" ] COUNT 
| ) [ ("myx"."x" in (uuid[char(32) "1aea00e5db6e0810b554fde31d961965"] as "%2"."%2", uuid[char(32) "1aea00e5db6e0810b554fde31d961966"] as "%3"."%3")) or ("myx"."y" = uuid[char(32) "1aea00e5db6e0810b554fde31d961967"]) ]
) [ "myx"."x", "myx"."y" ]
#insert into myx values ('1aea00e5db6e0810b554fde31d961965', '1aea00e5db6e0810b554fde31d961967');
[ 1	]
#select * from myx where x in ('1aea00e5db6e0810b554fde31d961966') or y is not null;
% sys.myx,	sys.myx # table_name
% x,	y # name
% uuid,	uuid # type
% 36,	36 # length
[ 1aea00e5-db6e-0810-b554-fde31d961965,	1aea00e5-db6e-0810-b554-fde31d961967	]
#select * from myx where x in ('1aea00e5db6e0810b554fde31d961966') or y is null;
% sys.myx,	sys.myx # table_name
% x,	y # name
% uuid,	uuid # type
% 36,	36 # length
#rollback;

# 11:57:38 >  
# 11:57:38 >  "Done."
# 11:57:38 >  

